llpc/lower/llpcSpirvLowerGlobal.cpp

```diff
@@ -1644,10 +1644,25 @@ void SpirvLowerGlobal::storeOutputMember(Type *outputTy, Value *storeValue, cons
 void SpirvLowerGlobal::lowerBufferBlock() {
   SmallVector<GlobalVariable *, 8> globalsToRemove;
 
+  // Represent the users of the global varibales, expect a bitCast, a GEP or a select used by GEPs
+  struct ReplaceInstsInfo {
+    BitCastInst *bitCastInst;                         // The user is a bitCast
+    SelectInst *selectInst;                           // The user is a select
```

Comment:
- What will happen if buffer ptr is used by multiple select instruction?
- What will happen if buffer ptr is used by phi instruction?  please let me know if it is handled in other place, since I didn't work on LLPC for pretty long time.

Reply:
> * What will happen if buffer ptr is used by multiple select instruction?

A ReplaceInstsInfo object is constructed per user and handled seperatedly. (in Line1698)
> * What will happen if buffer ptr is used by phi instruction?  please let me know if it is handled in other place, since I didn't work on LLPC for pretty long time.

I'm not sure if it can be used by phi instruction. There is no such case yet.
In the original version, there are only two cases are expected when the block is a array block as below:
@1 = external addspace(7) gloabl [2 x <{[10 x %llpc.array.element]}>], ...
case1: bitcast [2 x <{[10 x %llpc.array.element]}>] addrspace(7)* @1 to ... 
case2: getelementptr [4 x <{[10 x %llpc.array.element]}>], [4 x <{[10 x %llpc.array.element]}>] addrspace(7)* @1, ...
Now, there is a new case coming out. This change is to cover the case3.
@2 = external addspace(7) gloabl [2 x <{[10 x %llpc.array.element]}>], ...
case3: %2 = select i1 %1, [2 x <{ [10 x %llpc.array.element] }>] addrspace(7)* @1, [2 x <{ [10 x %llpc.array.element] }>] addrspace(7)* @2
%3 = getelementptr [4 x <{[10 x %llpc.array.element]}>], [4 x <{[10 x %llpc.array.element]}>] addrspace(7)* %2, ...